src/app/learnocaml_common.ml

```diff
@@ -708,11 +712,48 @@ let mouseover_toggle_signal elt sigvalue setter =
   in
   Manip.Ev.onmouseover elt hdl
 
+(* 
+
+   If a user has made no change to a solution for the exercise [id]
+   for 180 seconds, [check_valid_editor_state id] ensures that there is
+   no more recent version of this solution in the server. If this is
+   the case, the user is asked if we should download this solution
+   from the server.
+
+   This function reduces the risk of an involuntary overwriting of a
+   student solution when the solution is open in several clients.
+
+*)
+let check_valid_editor_state id =
+  let last_changed = ref (Unix.gettimeofday ()) in
+  fun update_content ->
+  let update_local_copy checking_time () =
+    match Learnocaml_local_storage.(retrieve (exercise_state id)) with
+    | { Answer.mtime; solution; _ } ->
+       if mtime > checking_time then (
+         if Js_utils.confirm
+              [%i "A more recent answer exists on the server. \
+                   Do you want to update the current one?"]
+         then
+           update_content solution;
+       );
+       Lwt.return ()
```

Comment:
Hi @yurug, just in case this is useful for implementing the front-end change (to turn the OK/Cancel JavaScript dialog into a 2+_choice learn-ocaml dialog), here is a working snippet:
```suggestion
         lwt_alert ~title:"Question"
           ~buttons: [ [%i "Fetch"], (fun () -> Lwt.return (update_content solution));
                       (* [%i "Keep Editing"], (fun () -> TODO ); *)
                       [%i "Abort"], (fun () -> Lwt.return_unit) ]
           [ H.p [H.txt [%i "A more recent answer exists on the server. \
                             Do you want to fetch the new version?"] ] ]
       ) else Lwt.return_unit
```

Reply:
Unfortunately, we cannot change the button labels of the confirmation dialog box (as far as I know).

Reply:
Done in d90944f